File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03352/p03352_num4_parsed.cpp
#include <iostream>

int main() {
    int X;
    std::cin >> X;
    
    int largestPower = 0;
    for (int b = 1; b <= X; ++b) {
        for (int p = 2; ; ++p) {
            int power = 1;
            for (int i = 0; i < p; ++i) {
                power *= b;
            }
            if (power > X) {
                break;
            } else {
                largestPower = power;
            }
        }
    }
    
    std::cout << largestPower << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03352/p03352_num4_parsed.py
import math

X = int(input())

largest_power = 1

for b in range(1, int(math.sqrt(X)) + 1):
    for p in range(2, int(math.log(X, b)) + 1):
        power = b ** p
        if power <= X:
            largest_power = max(largest_power, power)

print(largest_power)